homeassistant/components/litterrobot/vacuum.py

```diff
@@ -29,60 +38,71 @@
+        entities.append(
+            LitterRobotCleaner(robot=robot, entity_type=TYPE_LITTER_BOX, hub=hub)
+        )
 
     if entities:
         async_add_entities(entities, True)
 
 
-class LitterRobotCleaner(LitterRobotEntity, VacuumEntity):
+class LitterRobotCleaner(LitterRobotControlEntity, VacuumEntity):
     """Litter-Robot "Vacuum" Cleaner."""
 
     @property
-    def supported_features(self):
+    def supported_features(self) -> int:
         """Flag cleaner robot features that are supported."""
         return SUPPORT_LITTERROBOT
 
     @property
-    def state(self):
+    def state(self) -> str:
         """Return the state of the cleaner."""
         switcher = {
-            Robot.UnitStatus.CLEAN_CYCLE: STATE_CLEANING,
-            Robot.UnitStatus.EMPTY_CYCLE: STATE_CLEANING,
-            Robot.UnitStatus.CLEAN_CYCLE_COMPLETE: STATE_DOCKED,
-            Robot.UnitStatus.CAT_SENSOR_TIMING: STATE_DOCKED,
-            Robot.UnitStatus.DRAWER_FULL_1: STATE_DOCKED,
-            Robot.UnitStatus.DRAWER_FULL_2: STATE_DOCKED,
-            Robot.UnitStatus.READY: STATE_DOCKED,
-            Robot.UnitStatus.OFF: STATE_OFF,
+            LitterBoxStatus.CLEAN_CYCLE: STATE_CLEANING,
+            LitterBoxStatus.EMPTY_CYCLE: STATE_CLEANING,
+            LitterBoxStatus.CLEAN_CYCLE_COMPLETE: STATE_DOCKED,
+            LitterBoxStatus.CAT_SENSOR_TIMING: STATE_DOCKED,
+            LitterBoxStatus.DRAWER_FULL_1: STATE_DOCKED,
+            LitterBoxStatus.DRAWER_FULL_2: STATE_DOCKED,
+            LitterBoxStatus.READY: STATE_DOCKED,
+            LitterBoxStatus.CAT_SENSOR_INTERRUPTED: STATE_PAUSED,
+            LitterBoxStatus.OFF: STATE_OFF,
         }
 
-        return switcher.get(self.robot.unit_status, STATE_ERROR)
+        return switcher.get(self.robot.status, STATE_ERROR)
 
     @property
-    def status(self):
+    def status(self) -> str:
         """Return the status of the cleaner."""
-        return f"{self.robot.unit_status.label}{' (Sleeping)' if self.robot.is_sleeping else ''}"
+        return (
+            f"{self.robot.status.text}{' (Sleeping)' if self.robot.is_sleeping else ''}"
+        )
 
-    async def async_turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs) -> None:
         """Turn the cleaner on, starting a clean cycle."""
         await self.perform_action_and_refresh(self.robot.set_power_status, True)
 
-    async def async_turn_off(self, **kwargs):
+    async def async_turn_off(self, **kwargs) -> None:
         """Turn the unit off, stopping any cleaning in progress as is."""
         await self.perform_action_and_refresh(self.robot.set_power_status, False)
 
-    async def async_start(self):
+    async def async_start(self) -> None:
         """Start a clean cycle."""
         await self.perform_action_and_refresh(self.robot.start_cleaning)
 
-    async def async_send_command(self, command, params=None, **kwargs):
+    async def async_send_command(
```

Comment:
It seems like these might be easier to use as entity services since we can validate them.

https://developers.home-assistant.io/docs/dev_101_services/#entity-services

Reply:
Done